src/consumer.rs

```diff
@@ -173,63 +167,120 @@ impl JSValue<'_> for OffsetWrapper {
-                let element = tx.try_to_js(js_env)?;
-
-                // Update the array of aborted transactions;
-                js_env.set_element(aborted, element, index)?;
-            }
+#[node_bindgen]
+impl<'a> FetchablePartitionResponseWrapper {
+    #[node_bindgen(constructor)]
+    fn new() -> Self {
+        Self(None)
+    }
+    fn set_inner(&mut self, inner: Option<FetchablePartitionResponse<RecordSet>>) {
+        self.0 = inner;
+    }
+    #[node_bindgen(getter)]
+    fn error_code(&self) -> Option<i32> {
+        Some(self.0.as_ref()?.error_code as i32)
+    }
 
-            // set the aborted transactions;
-            response.set_property(ABORTED_KEY, aborted)?;
-        }
+    #[node_bindgen(getter)]
+    fn partition_index(&self) -> Option<i32> {
+        Some(self.0.as_ref()?.partition_index)
+    }
+
+    #[node_bindgen(getter)]
+    fn high_watermark(&self) -> Option<i64> {
+        Some(self.0.as_ref()?.high_watermark)
+    }
 
-        let record_set = RecordSetWrapper(self.0.records).try_to_js(js_env)?;
+    #[node_bindgen(getter)]
+    fn last_stable_offset(&self) -> Option<i64> {
+        Some(self.0.as_ref()?.last_stable_offset)
+    }
 
-        response.set_property(RECORDS_KEY, record_set)?;
+    #[node_bindgen(getter)]
+    fn log_start_offset(&self) -> Option<i64> {
+        Some(self.0.as_ref()?.log_start_offset)
+    }
+    #[node_bindgen(getter)]
+    fn aborted(&'a self) -> Vec<AbortedTransactionWrapper> {
+        let mut aborted_transactions: Vec<AbortedTransactionWrapper> = Vec::new();
 
-        response.try_to_js(js_env)
+        let inner = if let Some(inner) = self.0.as_ref() {
+            inner
+        } else {
+            return Vec::new();
+        };
+
+        if let Some(ref transactions) = inner.aborted {
+            for i in transactions {
+                aborted_transactions.push(AbortedTransactionWrapper(&i));
+            }
+        }
+        aborted_transactions
+    }
+
+    #[node_bindgen(getter)]
+    fn records(&'a self) -> Option<RecordSetWrapper> {
+        Some(RecordSetWrapper(&self.0.as_ref()?.records))
+    }
+    #[node_bindgen]
+    fn to_records(&'a self) -> Vec<String> {
+        let mut records = Vec::new();
+        let inner = if let Some(inner) = self.0.as_ref() {
+            inner
+        } else {
+            return Vec::new();
+        };
+        for batch in &inner.records.batches {
+            for record in &batch.records {
+                let value = record.get_value().inner_value_ref().clone();
+                let value = if let Some(value) = value {
+                    value
+                } else {
+                    continue;
+                };
+                if let Ok(value) = String::from_utf8(value) {
+                    records.push(value);
+                }
+            }
+        }
+        records
+    }
+}
+pub struct AbortedTransactionWrapper<'a>(&'a AbortedTransaction);
+impl TryIntoJs for AbortedTransactionWrapper<'_> {
```

Comment:
This is a work around for infinyon/node-bindgen#124. I'm not in love with it but want to ship this pr.